sops_test.go

```diff
@@ -1108,6 +1108,322 @@ func TestSetArrayNonLeaf(t *testing.T) {
+	assert.Equal(t, TreeBranch{
+	}, unset)
+}
+
+func TestUnsetKeyBranchLastLeaf(t *testing.T) {
+	branch := TreeBranch{
+		TreeItem{
+			Key: "foo",
+			Value: TreeBranch{
+				TreeItem{
+					Key: "bar",
+					Value: "bar",
+				},
+			},
+		},
+	}
+	unset, err := branch.Unset([]interface{}{"foo", "bar"})
+	assert.Equal(t, nil, err)
+	assert.Equal(t, TreeBranch{
+		TreeItem{
+			Key: "foo",
+			Value: TreeBranch{
+			},
+		},
+	}, unset)
+}
+
+func TestUnsetKeyArray(t *testing.T) {
+	branch := TreeBranch{
+		TreeItem{
+			Key: "foo",
+			Value: TreeBranch{
+				TreeItem{
+					Key: "bar",
+					Value: []interface{}{
+						TreeBranch{
+							TreeItem{
+								Key: "baz",
+								Value: "baz",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	unset, err := branch.Unset([]interface{}{"foo", "bar"})
+	assert.Equal(t, nil, err)
+	assert.Equal(t, TreeBranch{
+		TreeItem{
+			Key: "foo",
+			Value: TreeBranch{
+			},
+		},
+	}, unset)
+}
+
+func TestUnsetArrayItem(t *testing.T) {
+	branch := TreeBranch{
+		TreeItem{
+			Key: "foo",
+			Value: []interface{}{
+				TreeBranch{
+					TreeItem{
+						Key: "bar",
+						Value: "bar",
+					},
+				},
+				TreeBranch{
+					TreeItem{
+						Key: "barbar",
+						Value: "barbar",
+					},
+				},
+			},
+		},
+	}
+	unset, err := branch.Unset([]interface{}{"foo", 1})
+	assert.Equal(t, nil, err)
+	assert.Equal(t, TreeBranch{
+		TreeItem{
+			Key: "foo",
+			Value: []interface{}{
+				TreeBranch{
+					TreeItem{
+						Key: "bar",
+						Value: "bar",
+					},
+				},
+			},
+		},
+	}, unset)
+}
+
+func TestUnsetKeyInArrayItem(t *testing.T) {
+	branch := TreeBranch{
+		TreeItem{
+			Key: "foo",
+			Value: []interface{}{
+				TreeBranch{
+					TreeItem{
+						Key: "bar",
+						Value: "bar",
+					},
+					TreeItem{
+						Key: "barbar",
+						Value: "barbar",
+					},
+				},
+			},
+		},
+	}
+	unset, err := branch.Unset([]interface{}{"foo", 0, "barbar"})
+	assert.Equal(t, nil, err)
+	assert.Equal(t, TreeBranch{
+		TreeItem{
+			Key: "foo",
+			Value: []interface{}{
+				TreeBranch{
+					TreeItem{
+						Key: "bar",
+						Value: "bar",
+					},
+				},
+			},
+		},
+	}, unset)
+}
+
+func TestUnsetArrayLastItem(t *testing.T) {
+	branch := TreeBranch{
+		TreeItem{
+			Key: "foo",
+			Value: []interface{}{
+				TreeBranch{
+					TreeItem{
+						Key: "bar",
+						Value: "bar",
+					},
+				},
+			},
+		},
+	}
+	unset, err := branch.Unset([]interface{}{"foo", 0})
+	assert.Equal(t, nil, err)
+	assert.Equal(t, TreeBranch{
+		TreeItem{
+			Key: "foo",
+			Value: []interface{}{
+			},
+		},
+	}, unset)
+}
+
+func TestUnsetKeyNotFound(t *testing.T) {
+	branch := TreeBranch{
+		TreeItem{
+			Key: "foo",
+			Value: TreeBranch{
+				TreeItem{
+					Key: "bar",
+					Value: "bar",
+				},
+			},
+		},
+	}
+	unset, err := branch.Unset([]interface{}{"foo", "unknown"})
+	assert.Equal(t, err.(*SopsKeyNotFound).Key, "unknown")
+	if unset != nil {
+		t.Errorf("Unset result was not nil upon %s", err)
+	}
```

Comment:
I think you can also simply do
```suggestion
	assert.Nil(t, unset, "Unset result was not nil upon %s", err)
```
(https://github.com/stretchr/testify/blob/b074924938f86d417f1c9a845c7e8b0784d7f937/assert/assertions.go#L704-L715)

Same below.

Reply:
Thanks for the tip!